conformance/utils/kubernetes/helpers_test.go

```diff
@@ -0,0 +1,160 @@
+package kubernetes
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+func TestSupportedKindsMatch(t *testing.T) {
```

Comment:
```suggestion
func Test_listenersMatch(t *testing.T) {
```

Reply:
can make this change, but hope it doesnt mean all test cases associated with `listenersMatch` need to be handled in this PR :)